lib/auth.js

```diff
@@ -231,6 +234,47 @@ module.exports = new Model({
     }.bind(this));
   },
 
+  signInWithOAuthRedirect: function (args, window) {
+    // Check to see if we're running in the browser or not
+    if (typeof window === 'undefined') {
+      throw new Error('OAuth login is only available via the browser.')
+    } else if (!args.appId) {
+      throw new Error('You need to specify an app ID.');
+    } else if (!args.redirectUri) {
+      throw new Error('You need to specify a URI to redirect to after authorization.');
+    }
+
+    var originalArguments = arguments;
+    return this.checkCurrent().then(function(user) {
+      if (user) {
+        return this.signOut().then(function() {
+          return this.signIn.apply(this, originalArguments);
+        }.bind(this));
+      } else {
+        console.log('Signing in with OAuth (redirect)');
+        var url = [
+          config.host,
+          '/oauth/authorize',
+          '?response_type=token',
+          '&client_id=',
+          args.appId,
+          '&redirect_uri=',
+          args.redirectUri
+        ].join('');
+        window.location.href = url;
+      }
+    }.bind(this));
+  },
+
+  completeSignInWithOAuth: function (args) {
```

Comment:
Yeah, in the app - the bit that goes with it is zooniverse/zoo-react-starterify#37. All it does is parse the response, then redirect you to a given URL. I'm all for handling it inside this module, I'm not a big fan of tightly coupling this process to the app. 

Just did a quick bit of research - I could watch for the `[hashchange](https://html.spec.whatwg.org/multipage/indices.html#event-hashchange)` event and process that instead?